devolutions-gateway/src/api/fwd.rs

```diff
@@ -237,3 +254,288 @@ where
+            let client_stream = crate::ws::websocket_compat(client_ws);
+            let client_addr = source_addr;
+
+            async move {
+                info!(target = %request_uri, "WebSocket-WebSocket forwarding");
+
+                let info = SessionInfo::builder()
+                    .association_id(claims.jet_aid)
+                    .application_protocol(claims.jet_ap)
+                    .details(ConnectionModeDetails::Fwd {
+                        destination_host: claims.target_host,
+                    })
+                    .time_to_live(claims.jet_ttl)
+                    .recording_policy(claims.jet_rec)
+                    .build();
+
+                let result = Proxy::builder()
+                    .conf(conf)
+                    .session_info(info)
+                    .address_a(client_addr)
+                    .transport_a(client_stream)
+                    .address_b("8.8.8.8:8888".parse().expect("valid hardcoded value")) // NOTE: We don’t really use this address for anything else other than pcap recording, so it’s fine to use a placeholder for now.
+                    .transport_b(server_stream)
+                    .sessions(sessions)
+                    .subscriber_tx(subscriber_tx)
+                    .build()
+                    .select_dissector_and_forward()
+                    .instrument(span.clone())
+                    .await
+                    .context("encountered a failure during WebSocket traffic proxying");
+
+                if let Err(error) = result {
+                    span.in_scope(|| {
+                        error!(error = format!("{error:#}"), "WebSocket forwarding failure");
+                    });
+                }
+            }
+        })
+    } else {
+        // 4.a Plain HTTP request forwarding using reqwest.
+
+        let (parts, body) = request.into_parts();
+        let body = reqwest::Body::wrap_stream(body.into_data_stream());
+        let request = axum::http::Request::from_parts(parts, body);
+        let request = reqwest::Request::try_from(request).map_err(HttpError::internal().err())?;
+
+        debug!(?request);
+
+        info!(target = %request.url(), %dangerous_tls, "Forward HTTP request");
+
+        let client = if dangerous_tls { &*DANGEROUS_CLIENT } else { &*CLIENT };
+
+        let response = client.execute(request).await.map_err(HttpError::bad_gateway().err())?;
+
+        if let Err(error) = response.error_for_status_ref() {
+            info!(%error, host = claims.target_host.host(), "Service responded with an failure HTTP status code");
+        }
+
+        // 4.b Convert the response into the expected type and return it.
+
+        let response = Response::from(response);
+        let (parts, body) = response.into_parts();
+        let body = axum::body::Body::from_stream(body.into_data_stream());
+        Response::from_parts(parts, body)
+    };
+
+    return Ok(response);
+
+    // * Helpers //
```

Comment:
[nit][style] There was different section style in other places (`// -- local helper -- //`) is it intentional?

Reply:
Fair enough! I deviated to follow the 1., 2. (numbered) ... *, * (bullet) style, for a document-like readability. (Almost literate programming.)